server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationStatsService.java

```diff
@@ -39,6 +40,26 @@ public AllocationStatsService(
     }
 
     public Map<String, NodeAllocationStats> stats() {
-        return nodeAllocationStatsProvider.stats(clusterService.state(), clusterInfoService.getClusterInfo(), desiredBalanceSupplier.get());
+        var state = clusterService.state();
+        var stats = nodeAllocationStatsProvider.stats(
+            state.metadata(),
+            state.getRoutingNodes(),
+            clusterInfoService.getClusterInfo(),
+            desiredBalanceSupplier.get()
+        );
+        return stats.entrySet()
```

Comment:
the new `NodeWeight` didn't seem to make sense to add to the API (not useful and BWC trouble), but the stats provider logic is still shared.